src/mon/ConfigMonitor.cc

```diff
@@ -63,11 +67,28 @@ void ConfigMonitor::encode_pending(MonitorDBStore::TransactionRef t)
 
   // TODO: record changed sections (osd, mds.foo, rack:bar, ...)
 
+  string history = HISTORY_PREFIX + stringify(version+1) + "/";
+  {
+    bufferlist metabl;
+    ::encode(ceph_clock_now(), metabl);
+    ::encode(pending_description, metabl);
+    t->put(CONFIG_PREFIX, history, metabl);
+  }
   for (auto& p : pending) {
     string key = KEY_PREFIX + p.first;
+    auto q = current.find(p.first);
+    if (q != current.end()) {
+      if (p.second && *p.second == q->second) {
```

Comment:
Is this condition still needed now that pending is being cut down to only the bits that are changed in ConfigMonitor::prepare_command

Reply:
Usually not, but I want to make sure any other paths are captured.  Or we change the service later so that multiple updates get batched up together.